### ISSUE TITLE
在使用kubefate部署的FATE2.0.0版本上测试多个gpu运行的homo_nn出现“No found task output.”错误

### ISSUE THREAD

**Issue by linper1995** [FederatedAI/FATE#5623]
![微信截图_20240617224322](https://github.com/FederatedAI/FATE/assets/72566940/574ed650-61b0-4078-b871-9bb6f264d261)
![微信截图_20240617224330](https://github.com/FederatedAI/FATE/assets/72566940/5d943abd-1b93-42c0-894a-6e354d7aa484)
在使用kubefate部署的FATE2.0.0版本上测试多个gpu运行的homo_nn出现“No found task output.”错误，
**具体代码如下：**
from fate_test.utils import parse_summary_result
from fate_client.pipeline.utils import test_utils
from fate_client.pipeline.components.fate.evaluation import Evaluation
from fate_client.pipeline import FateFlowPipeline
from fate_client.pipeline.components.fate.nn.torch import nn, optim
from fate_client.pipeline.components.fate.nn.torch.base import Sequential
from fate_client.pipeline.components.fate.homo_nn import HomoNN, get_config_of_default_runner
from fate_client.pipeline.components.fate.nn.algo_params import TrainingArguments, FedAVGArguments
from fate_client.pipeline.components.fate import Reader
guest = "228"
host = "227"
arbiter = "227"

data_guest = "breast_homo_guest"
data_host = "breast_homo_host"
data_test =  "breast_homo_test"
idx = "id"
label_name = "y"
in_feat = 30
out_feat = 10
lr = 0.01
batch_size = 8
epochs = 2
guest_train_data = {"name": "breast_homo_guest", "namespace": "experiment"}
host_train_data = {"name": "breast_homo_host", "namespace": "experiment"}
test_data = {"name": "breast_homo_test", "namespace": "experiment"}
pipeline = FateFlowPipeline().set_parties(guest=guest, host=host, arbiter=arbiter)
basic_conf = {
    'task_device': {
        'type': 'gpu'
    },
    'task_run': {
        'eggroll': {
            'cores': 4  
        }
    }
}
pipeline.conf.set_all(**basic_conf)

reader_0 = Reader("reader_0", runtime_parties=dict(guest=guest, host=host))

reader_0.guest.task_parameters(
    namespace=guest_train_data['namespace'],
    name=guest_train_data["name"]
)
reader_0.hosts[0].task_parameters(
    namespace=host_train_data['namespace'],
    name=host_train_data["name"]
)

conf = get_config_of_default_runner(
    algo='fedavg',
    model=Sequential(
        nn.Linear(in_feat, out_feat),
        nn.ReLU(),
        nn.Linear(out_feat ,1),
        nn.Sigmoid()
    ), 
    loss=nn.BCELoss(),
    optimizer=optim.Adam(lr=lr),
    training_args=TrainingArguments(
        num_train_epochs=epochs, 
        per_device_train_batch_size=batch_size,
        seed=114514),
    fed_args=FedAVGArguments(),
    task_type='binary'
    )

homo_nn_0 = HomoNN(
        'nn_0',
        runner_conf=conf,
        train_data=reader_0.outputs["output_data"]
    )

homo_nn_1 = HomoNN(
    'nn_1',
    predict_model_input=homo_nn_0.outputs['train_model_output'],
    test_data=reader_0.outputs["output_data"]
)

evaluation_0 = Evaluation(
    'eval_0',
    runtime_parties=dict(guest=guest, host=host),
    metrics=['auc'],
    input_data=[homo_nn_1.outputs['predict_data_output']]
)

pipeline.add_tasks([reader_0, homo_nn_0, homo_nn_1, evaluation_0])
pipeline.compile()
pipeline.fit()

**错误信息：**
[INFO][2024-06-17 14:39:27,940][7910][trainer_base._client_send_parameters][line:521]: parameters is {'num_train_epochs': 2, 'max_steps': 30, 'num_update_steps_per_epoch': 15, 'epochs_trained': 0, 'steps_trained_in_current_epoch': 0, 'max_aggregation': 2, 'aggregate_freq': 1, 'aggregation_strategy': 'epoch', 'can_aggregate_loss': True}
310
[INFO][2024-06-17 14:39:37,658][6033][_wraps.run_component][line:168]: finish task, return code -7
311
[ERROR][2024-06-17 14:39:37,659][6033][_wraps.run][line:87]: No found task output.
